Add datetime formatting for ETA time.

hgaiser opened the pull request:
This PR changes the ETA while training from printing only the number of seconds, to printing it in a more human readable format (ie. `ETA: 9:17:04` for 9 hours, 17 minutes and 4 seconds, as opposed to `ETA: 33424s`).

fchollet commented:
I think this is valid when epochs are more than one hour long, but that is a rare case.

A reasonable format would be:

`1:23:45`(proposed)  if > 1 hour
`1:34` if >  1 min (note: no leading zero)
`3s` (current) otherwise.

hgaiser commented:
I'm currently training on COCO, my epochs are roughly 8 hours long :)

I was hoping to find the same thing as what you mentioned, implemented with `datetime`. I could only find custom defined formats, but then it might as well be a computed by hand since it's not rocket science. I updated this PR as per request.

A slight difference is that with `hours > 24`, the `datetime` approach would print something like `1 days, 9:17:04`. The current approach would print `33:17:04` but I'm okay with that.

fchollet commented on keras/utils/generic_utils.py:
```diff
@@ -328,7 +328,18 @@ def update(self, current, values=None, force=False):
+                elif minutes:
+                    eta_format = '%d:%02d' % (minutes, seconds)
+                else:
+                    eta_format = '%d' % seconds
```
Please add `s` for seconds.

hgaiser replied:
Done, nice catch.

fchollet commented on keras/utils/generic_utils.py:
```diff
@@ -328,7 +328,18 @@ def update(self, current, values=None, force=False):
                 eta = time_per_unit * (self.target - current)
-                info = ' - ETA: %0.fs' % eta
+                hours = eta // (60 * 60)
+                minutes = (eta % (60 * 60)) // 60
```
Replacing `60 * 60` with `3600` here and above would save quite a bit of computation, since this method is called a lot.

fchollet replied:
Actually, maybe it would be higher performance to have a switch:

`if eta > 3600` / `if eta > 60`

So that we only compute what we need.

hgaiser replied:
Ah right, used to c++ optimizations I guess. Changed as per request.

fchollet commented:
Reposting this:

Actually, maybe it would be higher performance to have a switch:

`if eta > 3600` / `if eta > 60`

So that we only compute what we need.

hgaiser commented:
Changed as per request.
